### ISSUE TITLE
difference in diff output between git diff and what's shown in sublime

### ISSUE THREAD

**Issue by canoeberry** [jisaacks/GitGutter#421]
## Description

Does it make sense that git diff would display "a delete of a single line followed by a bunch of added lines", but Sublime Text would display just "a bunch of changed lines"?

I originally was going to ask about performance of the plugin because it seemed really sluggish suddenly. But while filing this report, I discovered that if I made a small change to one line at the end of the set of changed lines, it changed all those "changed lines" back into a "delete of one line, an add of many others". And that itself might have been the cause of the sluggishness.

Regardless, the diff output always shows a delete followed by a lot of adds, and Sublime shows just changed lines.

## Support Info
- Sublime Text 3132
- Platform: osx
- Arch: x64
- GitGutter 1.7.1
- Install via PC: True
- git version 2.8.2
- mdpopups 1.13.1
- markdown 2.6.6
- pygments 2.1a0
- jinja2 2.8

**Comment by canoeberry** [comment]
Well - let me just come back and say that the issue of performance is another plugin, something I should have checked first.

But I am still curious about why the diff output is different.

**Comment by r-stein** [comment]
How do you get the information "a deleted line followed by added lines"? [here](https://github.com/jisaacks/GitGutter/blob/1.7.1/modules/handler.py#L439) is the code to parse the diff and the explanation why we are doing it like this is in the comment, but we are interested in improving it. So it would be nice if you had a suggestion for that.

**Comment by canoeberry** [comment]
The git diff in a Terminal window had a "-" before the deleted line and was displayed in red. And then the rest of the lines had "+" and displayed in green.

I've had a quick look at your code and it doesn't seem like magic so I think I can figure exactly what is going on and I will get back to you. Thanks for replying!

**Comment by canoeberry** [comment]
OK - after looking at your code I have decided that the cause of this difference is that you are passing -U0 and the default value with git is 3 lines of context. That explains everything.

I just never noticed this whole time which is ... kinda weird.

**Comment by r-stein** [comment]
Yes, but that shows the code before and after.
So if we have
Before:
```
A
B
```
and change it to
```
AA
BB
```
This will result in
```
- A
- B
+ AA
+ BB
```
On the other hand if we change it to 
```
AA
D
```
it will also result in
```
- A
- B
+ AA
+ D
```
---
In the first case I would mark both lines as change, but in the second I would the first line would be changed, then there would be a deleted block, and then there would be an added line. I don't think there is a good solution to decide whether it is similar enough to be a changed.

**Comment by canoeberry** [comment]
I think you just replied to a comment I deleted due to being confused. And I am still confused.

**Comment by r-stein** [comment]
I am just saying that the `+` and `-` signs are not as useful as you may think at the first moment =).

**Comment by canoeberry** [comment]
I was just saying, as your last comment came in, You know, I think that this whole Issue is bogus. I was very confused but now I am less confused. It's as though I thought Sublime was displaying the deleted lines but of course it doesn't.

Actually I think I know what happened to cause a bunch of "changed lines" (in sublime) to a bunch of "plus lines". I think I edited the ONE line that was in common between the before and after. And as soon as I changed that one line so that it was restored to its original value, it was no longer a change but rather an insert of a bunch of lines, terminated by that line in common. But as soon as I typed a single character on that line, it became part of a change and the "+" all disappeared and turned into change.

In other words, you just spent however long you spent dealing with this issue wasting your time educating me on how git gutter works. 

I learned a lot. But sorry for the waste of time!

**Comment by r-stein** [comment]
No, thats fine I like educating ;) The deleted lines are shown in the hover diff popup by-the-way.
